sim-lib/src/lib.rs

```diff
@@ -1077,8 +1179,20 @@ async fn track_payment_result(
         }
     };
 
-    if results.clone().send((payment, res)).await.is_err() {
-        log::debug!("Could not send payment result.");
+    select! {
+        biased;
+        _ = listener.clone() => {
+            log::debug!("Track payment received a shutdown signal")
+        },
+        send_payment_result = results.send((payment, res)) => {
+            match send_payment_result {
+                Ok(_) => {},
+                Err(e) => {
```

Comment:
Can we still use `is_err` here since we don't care about the `OK` branch in the match? LIkewise elsewhere in this PR.